server/migrations/community/d02961c7416c_add_project_member_table.py

```diff
@@ -0,0 +1,235 @@
+"""Add project member table
+
+Revision ID: d02961c7416c
+Revises: 1ab5b02ce532
+Create Date: 2024-10-31 15:20:52.833051
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "d02961c7416c"
+down_revision = "1ab5b02ce532"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    conn = op.get_bind()
+    conn.execute(sa.text("CREATE EXTENSION IF NOT EXISTS intarray;"))
+
+    op.create_table(
+        "project_member",
+        sa.Column("project_id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.Column("user_id", sa.Integer(), nullable=False),
+        sa.Column(
+            "role",
+            postgresql.ENUM("reader", "editor", "writer", "owner", name="project_role"),
+            nullable=False,
+        ),
+        sa.ForeignKeyConstraint(
+            ["project_id"],
+            ["project.id"],
+            name=op.f("fk_project_member_project_id_project"),
+            ondelete="CASCADE",
+        ),
+        sa.ForeignKeyConstraint(
+            ["user_id"],
+            ["user.id"],
+            name=op.f("fk_project_member_user_id_user"),
+            ondelete="CASCADE",
+        ),
+        sa.PrimaryKeyConstraint(
+            "project_id", "user_id", name=op.f("pk_project_member")
+        ),
+    )
+
+    op.add_column("project", sa.Column("public", sa.Boolean(), nullable=True))
+    op.create_index(op.f("ix_project_public"), "project", ["public"], unique=False)
+
+    data_upgrade()
+
+    op.drop_table("project_access")
+    conn.execute(sa.text("DROP EXTENSION IF EXISTS intarray;"))
+
+
+def downgrade():
+    op.create_table(
+        "project_access",
+        sa.Column("public", sa.BOOLEAN(), autoincrement=False, nullable=True),
+        sa.Column(
+            "owners",
+            postgresql.ARRAY(sa.INTEGER()),
+            server_default=sa.text("'{}'::integer[]"),
```

Comment:
This is different to the schema we run right now in production (we don't use a default)

Reply:
Discussed and this should not matter